core/src/main/java/nu/yona/server/subscriptions/entities/BuddyConnectMessage.java

```diff
@@ -4,49 +4,122 @@
  *******************************************************************************/
 package nu.yona.server.subscriptions.entities;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
+import javax.persistence.Column;
 import javax.persistence.Entity;
 import javax.persistence.Transient;
 
 import nu.yona.server.crypto.seckey.SecretKeyUtil;
+import nu.yona.server.device.entities.DeviceBase;
+import nu.yona.server.device.entities.UserDevice;
 import nu.yona.server.messaging.entities.BuddyMessage;
 
 @Entity
 public abstract class BuddyConnectMessage extends BuddyMessage
 {
+	private static final int UUID_LENGTH = 36;
+	private static final int BOOLEAN_LENGTH = 5;
+	private static final int MAX_NUM_OF_DEVICES = 10;
+	private static final double ENCRYPTION_OVERHEAD_FACTOR = 1.2; // This is a guess
+
 	@Transient
 	private UUID buddyId;
 	private byte[] buddyIdCiphertext;
 
+	@Transient
+	private String deviceNames;
+	@Column(length = (int) ((DeviceBase.MAX_NAME_LENGTH + 1) * MAX_NUM_OF_DEVICES * ENCRYPTION_OVERHEAD_FACTOR))
+	private byte[] deviceNamesCiphertext;
+
+	@Transient
+	private String deviceAnonymizedIds;
+	@Column(length = (int) ((UUID_LENGTH + 1) * MAX_NUM_OF_DEVICES * ENCRYPTION_OVERHEAD_FACTOR))
+	private byte[] deviceAnonymizedIdsCiphertext;
+
+	@Transient
+	private String deviceVpnConnectionStatuses;
+	@Column(length = (int) ((BOOLEAN_LENGTH + 1) * MAX_NUM_OF_DEVICES * ENCRYPTION_OVERHEAD_FACTOR))
+	private byte[] deviceVpnConnectionStatusesCiphertext;
+
 	// Default constructor is required for JPA
 	protected BuddyConnectMessage()
 	{
 		super();
 	}
 
-	protected BuddyConnectMessage(BuddyInfoParameters buddyInfoParameters, String message, UUID buddyId)
+	protected BuddyConnectMessage(BuddyInfoParameters buddyInfoParameters, String message, UUID buddyId, Set<UserDevice> devices)
 	{
 		super(buddyInfoParameters, message);
 		this.buddyId = buddyId;
+		List<UserDevice> sortedDevices = devices.stream()
+				.sorted((d1, d2) -> d1.getDeviceAnonymizedId().compareTo(d2.getDeviceAnonymizedId()))
```

Comment:
Isn't the sorting a bit superfluous here? Why not just convert it to a list?

Reply:
You're right. The order doesn't matter, as long as it's fixed. I'l update it.